crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1915,12 +1927,21 @@ impl<'db> TypeInferenceBuilder<'db> {
 
         if let Some(value) = value.as_deref() {
             let value_ty = self.infer_expression(value);
-            self.add_declaration_with_binding(
-                assignment.into(),
-                definition,
-                annotation_ty,
-                value_ty,
-            );
+            if self.file().is_stub(self.db().upcast()) {
```

Comment:
It seems we've forgotten the condition "and the `value` expression is a literal `...`" here. If that doesn't fail any test, we should definitely add a test with e.g. `x: int = 1` in a stub file showing we still infer `Literal[1]` for `x`, not `Unknown`.

Reply:
Yes it was not failing .Added a new test for it in `mdtest/assignment/annotations.md`.